ambry-commons/src/test/java/com.github.ambry.commons/BlobIdTest.java

```diff
@@ -245,7 +256,7 @@ public void testCrafting() throws Exception {
     assertEquals("Containers should match", crafted.getContainerId(), craftedAgain.getContainerId());
     assertEquals("The id string should match", crafted.getID(), craftedAgain.getID());
 
-    if (version == BLOB_ID_V3) {
+    if (version >= BLOB_ID_V3) {
```

Comment:
this test is not version specific, so the if condition was meant to be true only once.